src/lexer/Lexer.ts

```diff
@@ -476,6 +476,10 @@ export class Lexer {
 
             if (numberOfDigits >= 10 && designator !== "&") {
                 // numeric literals over 10 digits with no type designator are implicitly Doubles
+                if (designator === "#") {
+                    // but it's legal to explicitly designate it a double anyway
```

Comment:
Does swapping below `else if` with `if` on line 477 fixes the issue without needing to add this extra `if` block?
Also, not part of this change but line 488 looks like a dupe of 473

Reply:
Oooh it might!  Let me try.

Reply:
Good catch re: the duplicate line.  `advance()` changes the value of `current` so _technically_ we should be grabbing the text between `start` and the new `current`.  In practice it doesn't matter, because we pass that to JavaScript's `Float.fromString()` which ignores the trailing `#` that the second `source.slice()` returns anyway 🤷 Removing that too!